File: frontend/src/api/knowledgePoints.ts
import { apiFetch } from '@/api/client'
import type { KnowledgePoint } from '@/types/api'

/** 知识点目录（id / level1 / level2 中文名），用于前端把 id 显示成中文。 */
export const listKnowledgePoints = () =>
  apiFetch<KnowledgePoint[]>('/knowledge-points')

File: backend/api/knowledge_points.py
from __future__ import annotations

from fastapi import APIRouter, Depends

from backend.deps import current_user
from backend.schemas import User
from shared import storage
from shared.schemas import KnowledgePoint

router = APIRouter(prefix="/knowledge-points", tags=["knowledge-points"])


@router.get("", response_model=list[KnowledgePoint])
async def list_knowledge_points(_: User = Depends(current_user)) -> list[KnowledgePoint]:
    """Full KP catalog (id / level1 / level2 中文名) for frontend display."""
    return storage.list_knowledge_points()
